test/screenshot/spec/mdc-dialog/fixture.js

```diff
@@ -41,16 +41,21 @@ window.mdc.testFixture.fontsLoaded.then(() => {
     const shouldScrollToBottom = dialogEl.classList.contains('test-dialog--scroll-to-bottom');
     if (contentEl && shouldScrollToBottom) {
       const scrollToBottom = () => {
-        const tryToScroll = () => {
-          contentEl.scrollTop = contentEl.scrollHeight;
-          if (contentEl.scrollTop === 0) {
-            requestAnimationFrame(tryToScroll);
-          }
-        };
-        tryToScroll();
+        contentEl.scrollTop = contentEl.scrollHeight;
+        const lastChild = contentEl.children[contentEl.children.length - 1];
+        if (lastChild && lastChild.scrollIntoView) {
+          lastChild.scrollIntoView(/* topAligned */ false);
+        }
       };
-      dialog.listen(strings.OPENING_EVENT, scrollToBottom);
-      dialog.listen(strings.OPENED_EVENT, scrollToBottom);
+
+      const handleOpenEvent = () => {
+        for (let ms = 0; ms < 500; ms += 50) {
```

Comment:
What's the point of doing this incrementally instead of just waiting until the final interval? (Also, is this final interval longer than the system actually waits to take screenshots?)

Reply:
Because the screenshot could potentially get captured _before_ the final interval. This way it covers all the bases.

Reply:
I'm still confused, because if the screenshots are already captured, what's the point of still looping here?

Like, if the point of this loop is to set up the test page correctly for the screenshot, it seems like the uncertainty of when the screenshot is taken relative to where this loop currently is defeats the point. Either it's scrolled to the bottom at the right time, or it's not, no sense in trying to do it 10 times.

Reply:
It's not only the uncertainty of when the screenshot gets taken - it's also the uncertainty of exactly when the browser finishes reflowing after the `--scrollable` class gets added by JS.

When `--scrollable` gets added, it affects the `__content`'s `scrollHeight`, which means we might need to scroll again (depending on the browser). This is what was previously happening in Firefox.

I'll just remove the loop for now. We can always add it back later if necessary.